### DISCUSSION TITLE
Communication with WinCC Unified

### DISCUSSION THREAD

**Discussion by simonkampe** [simatic-ax/.discussions#25] [Q&A] [answered]
How would one use Simatic AX with a HMI (WinCC Unified)? Currently we use DBs to aggregate the information received/sent to the HMI, but I cannot find any examples on communication between these technologies.

**Comment by BeckerStS** [comment, marked as answer]
Hi @simonkampe ,

this is a perfect question for our simatic ax [forum](https://community.mendix.com/link/space/simatic-ax), where you find general questions product related. If you don't mind pls. shoot your question over there in addition.

**Anyway:** 
Currently SIMATIC AX capabilities to expose variables to external/internal are limited. There are no integraded / internal connection mechanism as of now but you can declare variables which should be exposed and use external mechanism. 

For any external client who wants to get in touch with the variables from the plc1500 runtime (ax loaded) we recommend creating an opcua serverinterface.
- activate the opcua server at the hardware configuration
- declare variables first with the {S7.extern=...} followed by the {OpcUa = ...} pragma, which need to be considered to be exposed, example:
```
    {OpcUa = ReadWrite}
    {S7.extern=ReadWrite}
    VAR_GLOBAL
        var1 : BOOL;
        var2 : typeA;
    END_VAR
``` 
- install OSCr 
```typescript
   apax add @ax/opcua-server-config
``` 
- compile your sw (stc)  > compile your hardware config (hwc) > load hardware (hwl) > load software (sld) > install serverinterface (oscr)
- export the serverinterface nodeset.xml using OSCr
- share the serverinterface xml with opcua clients (in this case you wincc unified runtime project), alternative browse live for the the opcua serverinterface to create your variables-mapping in the desired tool

**On a sitenote:**
Any variables exposed via the **S7.extern** pragma this way will be accessable by any client tools who can establish an non integrated S7 connection (incl. browsing) ... for example NX MCD. 
If you know the FQN of the variables on the runtime you may can make other mechanism work as well ... this is at own risk and and may subject to change, be aware.

**Reply by sjuergen** [reply]
@simonkampe pleas keep in mind, thatr we've the Mndix forum , where questions are better located:

https://community.mendix.com/link/space/simatic-ax
